internal/provider/resource_gitlab_project_runner.go

```diff
@@ -0,0 +1,120 @@
+package provider
+
+import (
+	"log"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+var _ = registerResource("gitlab_project_runner", func() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGitlabProjectRunnerCreate,
+		Read:   resourceGitlabProjectRunnerRead,
+		Delete: resourceGitlabProjectRunnerDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"project": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+			"runner_id": {
+				Type:     schema.TypeInt,
+				ForceNew: true,
+				Required: true,
+			},
+		},
+	}
+})
+
+func resourceGitlabProjectRunnerCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*gitlab.Client)
+	projectID := d.Get("project").(string)
+	runnerID := d.Get("runner_id").(int)
+	options := &gitlab.EnableProjectRunnerOptions{
+		RunnerID: runnerID,
+	}
+
+	log.Printf("[DEBUG] create gitlab project runner %s/%v", projectID, runnerID)
+
+	_, resp, err := client.Runners.EnableProjectRunner(projectID, options)
+	// TODO: Implement proper error handling with parsin of response message
+	// Ignore error if response contains "{message: {assign_to: [Validation failed: Runner has already been taken]}}"
+	if err != nil && resp.StatusCode != 400 {
+		return err
+	}
+
+	runnerIDString := strconv.Itoa(runnerID)
+	d.SetId(buildTwoPartID(&projectID, &runnerIDString))
+
+	return resourceGitlabProjectRunnerRead(d, meta)
+}
+
+func resourceGitlabProjectRunnerRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*gitlab.Client)
+	runnerType := "project_type"
+	projectID, runnerID, err := projectAndRunnerFromID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	options := &gitlab.ListProjectRunnersOptions{
+		Type: &runnerType,
+	}
+
+	log.Printf("[DEBUG] read gitlab project runner %s/%v", projectID, runnerID)
+
+	runners, _, err := client.Runners.ListProjectRunners(projectID, options)
+	if err != nil {
+		log.Printf("[DEBUG] failed to read gitlab project runner %s/%v: %s", projectID, runnerID, err)
+		d.SetId("")
+		return nil
+
+	}
+
+	found := false
+	for _, runner := range runners {
+		if runner.ID == runnerID {
+			found = true
+			break
+		}
+	}
+	if !found {
+		log.Printf("[WARN] removing project runner: %v from state because it no longer exists in gitlab", runnerID)
+		d.SetId("")
+	}
+
+	return nil
+}
+
+func projectAndRunnerFromID(id string) (string, int, error) {
+	projectID, runnerIDString, err := parseTwoPartID(id)
+	runnerID, e := strconv.Atoi(runnerIDString)
+	if err != nil {
+		e = err
+	}
+	if e != nil {
+		log.Printf("[WARN] cannot get project and runner ids from input: %v", id)
+	}
+	return projectID, runnerID, e
```

Comment:
It seems like you compress these two `if` statement, right?

```suggestion
	if err != nil {
		log.Printf("[WARN] cannot get project and runner ids from input: %v", id)
	}
	return projectID, runnerID, err
```

Reply:
It's actually conflating two separate errors, I will improve it to handle both cases